xarray/tests/test_variable.py

```diff
@@ -2939,6 +2955,23 @@ def test_LazilyIndexedArray(self):
         )
         self.check_orthogonal_indexing(v)
 
+    @pytest.mark.asyncio
+    async def test_lazy_async_indexing(self) -> None:
+        v = Variable(dims=("x", "y"), data=LazilyIndexedArray(self.d))
+        await self.check_orthogonal_async_indexing(v)
```

Comment:
This is fine, but we could combine the sync and async checks in one async function and just use that everywhere in this file.

Reply:
Done in https://github.com/pydata/xarray/pull/10327/commits/4f40792a27849508131465a9e341d359934b0ded, though now I'm a little worried that this pattern

```python
    async def check_orthogonal_indexing(self, v):
        expected = self.d[[8, 3]][:, [2, 1]]

        result = v.isel(x=[8, 3], y=[2, 1])
        assert np.allclose(result, expected)

        result = await v.isel(x=[8, 3], y=[2, 1]).load_async()
        assert np.allclose(result, expected)
```

might be automatically passing the second assertion by still being in-memory after the first assert?

Reply:
You could parametrize the test over async and non async calls

```python
@pytest.mark.parametrize("use_async", [True, False])
async def check_orthogonal_indexing(self, v, use_async):
        expected = self.d[[8, 3]][:, [2, 1]]

        if use_sync:
            result = await v.isel(x=[8, 3], y=[2, 1]).load_async()
        else:
            result = v.isel(x=[8, 3], y=[2, 1])
        assert np.allclose(result, expected)
```

Reply:
Could use `assert not v._in_memory` to be really sure.

Reply:
I went with the parametrization idea, though the syntax has to be a little messier because you can't parametrize normal functions in pytest. https://github.com/pydata/xarray/pull/10327/commits/a074a25e2cd74c6e5069519ab01016f563d8b11d